docs/src/how_it_works.md

```diff
@@ -0,0 +1,77 @@
+```math
+\iiint f(\bar{r}) ~ \text{d}V
+```
+
+Integrals like this are often solved manually by selecting an appropriate coordinate system and limits that neatly represent the integration domain, e.g.
+```math
+\int_0^{\pi} \int_0^{2\pi} \int_0^{5} f(\bar{r}) ~ \text{d}\rho~\text{d}\theta~\text{d}\phi
+```
+
+This works great for simple geometries, but requires integration code that is geometry-specific. This package, by contrast, leverages parametric functions defined in Meshes.jl and differential forms to define integral methods that are general solutions for all geometries.
+
+## Parametric Functions
+
+Every supported `Meshes.Geometry` type is defined as having a parametric function that maps from a local parametric coordinate system to every point on the geometry. Curve-like geometries will have a single parametric dimension, surfaces will have two dimensions, and volumes will have three dimensions; this can be checked for a particular geometry via `Meshes.paramdim(geometry)`.
+
+For consistency across geometry types, these parametric functions are defined to always take coordinates inside a normalized range $[0,1]$. In the example case of `ball`, Meshes.jl defines a parametric function mapped in normalized spherical coordinates $(t_\rho, ~t_\theta, ~t_\phi)$. We find, then:
+```julia
+Meshes.paramdim(ball) == 3    # a volume
+
+ball(tρ, tθ, tφ)    # for args in range [0, 1], maps to a corresponding Meshes.Point
+
+ball(0, tθ, tφ) == center
+```
+
+In effect, we can now use the geometry itself as a function that maps from three normalized ($0 \le t \le 1$) arguments to every point on the geometry. For the sake of generalization, let this parametric function be called $g$.
+```math
+\text{g}(t_1,~t_2,~t_3) ~\mapsto~ \text{Point}\big[ x, ~y, ~z \big]  
+```
+
+
+## Differential Forms
+
+Using differential forms, the general solution for integrating a geometry with three parametric dimensions ($t_1$, $t_2$, and $t_3$) is
+```math
+\iiint f(r̄) ~ \text{d}V = \iiint f(\bar{r}) ~ \bar{\text{d}t_1} \wedge \bar{\text{d}t_2} \wedge \bar{\text{d}t_3}
+```
+
+This resultant differential (volume) element is formed at each point in the integration domain by taking [the Jacobian](https://en.wikipedia.org/wiki/Jacobian_matrix_and_determinant) of the parametric function.
+```math
+\mathbf{J}_f = \begin{bmatrix} \bar{\text{d}t_1} & \bar{\text{d}t_2} & \bar{\text{d}t_3} \end{bmatrix}
+```
+where
+```math
+\bar{\text{d}t_n} = \frac{\partial}{\partial t_n} ~ \text{g}(t_1,~t_2,~t_3)
+```
+
+Each of these partial derivatives is a vector representing the direction that changing each parametric function argument will move the resultant point. The differential element ($E$) size is then calculated using geometric algebra as the magnitude of the exterior product ($\wedge$) of these three vectors.
+```math
+E(t_1,~t_2,~t_3) = \left\| \bar{\text{d}t_1} \wedge \bar{\text{d}t_2} \wedge \bar{\text{d}t_3} \right\|
```

Comment:
Maybe a stupid question, which I also could have asked before:
Isn't this not the absolute value of the determinant of the Jacobian of g? For example in 3D:
```julia
julia> J = rand(3, 3)
3×3 Matrix{Float64}:
 0.000905113  0.494132  0.449875
 0.417646     0.122622  0.532629
 0.408989     0.498596  0.854013

julia> abs(det(J))
0.0023697470337338046

julia> abs((J[:, 1] × J[:, 2]) ⋅ J[:, 3])
0.002369747033733771

julia> using CliffordNumbers

julia> J_kvecs = [CliffordNumbers.KVector{1, VGA(3)}(J[:, 1]...), CliffordNumbers.KVector{1, VGA(3)}(J[:, 2]...), CliffordNumbers.KVector{1, VGA(3)}(J[:, 3]...)]
3-element Vector{KVector{1, VGA(3), Float64, 3}}:
 KVector{1, VGA(3), Float64}(0.0009051134514089787, 0.41764622933744966, 0.4089885721993082)
   KVector{1, VGA(3), Float64}(0.49413189336803354, 0.12262234379158543, 0.4985961890818985)
    KVector{1, VGA(3), Float64}(0.44987481501348114, 0.5326286546861126, 0.8540130191622195)

julia> foldl(∧, J_kvecs)
1-element KVector{3, VGA(3), Float64}:
0.002369747033733785e₁e₂e₃
```
which means all of the three methods (`abs(det(J))` as I propose, the explicit dimension-dependent version we used before switching to CliffordNumbers.jl and the CliffordNumbers.jl version we use now) give the same result. Similar for 2D:
```julia
julia> J2 = rand(2, 2)
2×2 Matrix{Float64}:
 0.572102  0.068292
 0.937427  0.505511

julia> abs(det(J2))
0.2251849308605629

julia> using Meshes

julia> norm(Vec(J2[:, 1]...) × Vec(J2[:, 2]...))
0.22518493086056293 m^2

julia> J_kvecs = [CliffordNumbers.KVector{1, VGA(2)}(J2[:, 1]...), CliffordNumbers.KVector{1, VGA(2)}(J2[:, 2]...)]
2-element Vector{KVector{1, VGA(2), Float64, 2}}:
  KVector{1, VGA(2), Float64}(0.5721015978943291, 0.9374270118081014)
 KVector{1, VGA(2), Float64}(0.06829195845197433, 0.5055110114482464)

julia> foldl(∧, J_kvecs)
1-element KVector{2, VGA(2), Float64}:
0.22518493086056293e₁e₂
```

Do we actually need CliffordNumbers.jl? If `abs(det(J))` gives the same result, this would reduce complexity and resembles the integration by substitution formula (https://en.wikipedia.org/wiki/Integration_by_substitution#Substitution_for_multiple_variables) closely, which nothing really else is the whole story about parametric functions about, isn't it? I mean, the CliffordNumbers.jl solution is really elegant, but so is `abs(det(J))`, isn't it? And it involves less complicated maths, such that people with knowledge from basic multidimensional analysis and integrals (change of variables in multi-D) can understand what happens without needing to have a deeper knowledge about Clifford numbers. Or am I missing something?
Sorry for getting a bit off-topic of this PR, but since this PR also contains the documentation part, I wanted to mention it here.

Reply:
This PR is mainly focused on implementing `DifferentiationMethod`, and the doc updates were meant to shed some light on why derivatives are even being calculated.

That said... I actually hadn't realized that `abs(det(J::Matrix))` was mathematically equivalent to the current `abs(foldl(∧, J::NTuple{N,KVector})`. I'm really happy with the elegance of #114, but if we can develop a determinant-based solution that is similarly elegant and performant, then it's probably superior in practice for the reasons you've mentioned. I suspect that solution could look something like
```julia
using StaticArrays

function _SMatrix(
    vecs::NTuple{N, T}
) where {N, T <: Meshes.Vec}
    return SMatrix{N, N}((vecs...)...)
end

differential(geometry, ts, diff_method) = jacobian(geometry, ts, diff_method) |> _SMatrix |> det |> abs
```

I propose we document this is a new enhancement Issue and, if a PR for it pans out as expected, we could tweak this docs page again to explain the new method.

Reply:
I just realized that the Jacobian doesn't need to be square, i.e. it is not square if the `paramdim` of the `geometry` is not its `embeddim`. In this case the determinant solution would not work, while the `foldl(∧, J_kvecs)` works also in this case.

Reply:
Damn, I was pretty excited about this but you're correct. E.g. a `Segment` embedded in 3D space would have a Jacobian tuple consisting of a single 3D `Meshes.Vec`.

Reply:
Maybe still worth documenting the idea along with the drawback, just in case we want to revisit it later with fresh ideas.

Reply:
Agreed, but it's also interesting to see that the geometric algebra solution seems to be kind of a generalization of the classical substitution formula for transformations mapping between domains of different dimensions. Definitely, learned something by thinking about this :smiley:

Reply:
I'm just now reading that link and was surprised to see how directly-comparable that first theorem was to what we've implemented. I've definitely learned a lot along the way while building this package.